tools/tpm2_sign.c

```diff
@@ -135,36 +143,38 @@ static bool init(TSS2_SYS_CONTEXT *sapi_context) {
     }
 
     /*
-     * Process the msg file
+     * Process the msg file if needed
      */
-    unsigned long file_size;
-    bool result = files_get_file_size_path(ctx.inMsgFileName, &file_size);
-    if (!result) {
-        return false;
-    }
-    if (file_size == 0) {
-        LOG_ERR("The message file \"%s\" is empty!", ctx.inMsgFileName);
-        return false;
-    }
-
-    if (file_size > 0xffff) {
-        LOG_ERR(
-                "The message file was longer than a 16 bit length, got: %lu, expected less than: %d!",
-                file_size, 0x10000);
-        return false;
-    }
-
-    ctx.msg = (BYTE*) calloc(required_argument, file_size);
-    if (!ctx.msg) {
-        LOG_ERR("oom");
-        return false;
-    }
-
-    ctx.length = file_size;
-    result = files_load_bytes_from_path(ctx.inMsgFileName, ctx.msg, &ctx.length);
-    if (!result) {
-        free(ctx.msg);
-        return false;
+    if (ctx.flags.m && !ctx.flags.D) {
+      unsigned long file_size;
+      bool result = files_get_file_size_path(ctx.inMsgFileName, &file_size);
+      if (!result) {
+          return false;
+      }
+      if (file_size == 0) {
+          LOG_ERR("The message file \"%s\" is empty!", ctx.inMsgFileName);
+          return false;
+      }
+
+      if (file_size > UINT16_MAX) {
+          LOG_ERR(
+                  "The message file was longer than a 16 bit length, got: %lu, expected less than: %d!",
+                  file_size, 0x10000);
```

Comment:
Close (UINT16_MAX + 1) here or change the wording to "less than or equal to" and use UINT16_MAX, or just drop it and leave it less than 16 bits in length, and show what size you read, folks can figure that out from there.

Reply:
Can I change the message to "The message file is too big, got: %lu, expected less than: %u bytes" ?

"Longer than 16 bit length" seems weird to me :)